### PR TITLE
loki-canary: Fix resource leak when using client-side certs

### PR DESCRIPTION
**Pull request by chodges15**
**What this PR does / why we need it**:
When running canary with client side certificates (introduced in #6310), a new `http.Transport` is being created for each request 🤦‍♂️. As a result, connections are left open until they timeout an hour later. This quickly causes resource exhaustion and errors like `dial tcp 1.1.1.1:1111: connect: cannot assign requested address`. 

This change will reuse the same transport for all requests just like `http.DefaultClient` is reused when not using client certs.

Tested by running loki-canary for 5 days and observing with netstat a constant number of TCP connections to the querier.

### CODE REVIEW

**Review by DylanGuedes — APPROVED** [review]
lgtm

File: pkg/canary/reader/reader.go [hunk 1]
```diff
@@ -92,6 +93,19 @@ func NewReader(writer io.Writer,
...
+
+	// http.DefaultClient will be used in the case that the connection to Loki is http or TLS without client certs.
+	httpClient := http.DefaultClient
+	if tlsConfig != nil {
+		// For the mTLS case, use a http.Client configured with the client side certificates.
+		rt, err := config.NewTLSRoundTripper(tlsConfig, caFile, func(tls *tls.Config) (http.RoundTripper, error) {
+			return &http.Transport{TLSClientConfig: tls}, nil
+		})
+		if err != nil {
+			panic(fmt.Sprintf("Failed to create HTTPS transport with TLS config: %v", err))
```
Comment (dannykopping) [inline pkg/canary/reader/reader.go:105]: I don't think we should be using `panic`. Can we handle this in a less destructive way?

**Review by dannykopping — CHANGES_REQUESTED** [review]
LGTM, thanks for this! Let's please find a way to not use `panic`

File: pkg/canary/reader/reader.go (hunk 1 above)
Comment (chodges15) [reply pkg/canary/reader/reader.go]: Good point, there's no reason for a stack trace to come out here. I've kept this as a fatal error that will use `os.Exit` instead.

**Pull request merged**